lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,9 +166,64 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
-    return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
+
+  if (ppResult == nullptr || *ppResult == nullptr)
+    return S_OK;
+
+  HRESULT HR;
+  (*ppResult)->GetStatus(&HR);
+  if (FAILED(HR))
+    return HR;
```

Comment:
This changes the protocol of the function.  The HRESULT returned from the function indicates whether an error so fatal has occurred that it is unable to even write error details to a `IDxcOperationResult **ppResult` object.  This is the pattern used in our APIs that use `IDxcOperationResult` or newer `IDxcResult` interfaces.  If this returns a failing HR, the caller shouldn't even attempt to look at `ppResult`.

This changes it to return a failing HR if the validator did, then `ppResult` would need to be looked at to see the validation failure details.